zaius/s3/utils.py

```diff
@@ -0,0 +1,63 @@
+import boto3
+import os
+from functools import partial
+from multiprocessing import Pool, cpu_count
+
+import zaius.auth as auth
+
+class S3Client():
+
+    def __init__(self, auth_struct=None):
+        self.auth = auth_struct if auth_struct is not None else auth.default()
+        self.client = init_s3_client(self.auth)
```

Comment:
it feels like this `self.client` should be hidden for users of this class.
also this client is not used in any of the functions below (?)